### DISCUSSION TITLE
Android fit File upload

### DISCUSSION THREAD

**Discussion by NickSchouten** [dvmarinoff/Auuki#180] [Q&A]
Hey Guys, 

First of all what a great tool. I love it. Been using it for a year now.
However something has changed on my phone. I used to be able to download a .Fit file and use the default android Open With call to import it into Garmin connect, Komoot and Wahoo. I used the latter. 

However since new year that is no longer possible. All three claim a corrupted file. I know the .fit is not corrupted because manually importing to strava on my laptop works and the old .fit files, that I was once able to import into wahoo, now also report corrupted.

Does anybody know why or maybe what flow do you guys use to do upload your fits to any service directly from android.

**Comment by dvmarinoff** [comment]
Hi, thanks for the report!

I did a complete overhaul of the .FIT code during the holidays. There were some bugs that I've removed since, but this one was unknown. Was able to do only a very quick test for now. Did a 3 min ride on my Android phone and uploaded the .FIT file to Garmin Connect, it uploaded without error, and forwarded the file to Strava and intervals.icu, again successfully.

My app version is: 0.1.43
Samsung Galaxy S9
Google Chrome

I'll do a longer test for tomorrow and will double check the file with the SDK tools.

<img width="320" src="https://github.com/dvmarinoff/Flux/assets/7523832/7241a353-4153-4cab-86b9-5830ced4700c" />

**Comment by dvmarinoff** [comment]
On a second thought can you share one of those problematic .FIT files?

**Reply by NickSchouten** [reply]
Sure thing, I made a private repo https://github.com/NickSchouten/temp_private_fit_file_share
That you should have access to. It has 3 fit files. 

But again even the files from early december, which did work at one point, now no longer import directly with the OpenWith Android callback.

I you want I can share them too.
